### PR TITLE
Add some missing error handling.

### LINKED ISSUE #364: makeInformer watch stops receiving updates for "/api/v1/configmaps"
```
  const kc = new k8s.KubeConfig();
  kc.loadFromDefault();

  const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
  const listFn = () => k8sApi.listConfigMapForAllNamespaces();
  const informer = k8s.makeInformer(kc, "/api/v1/configmaps", listFn);

  informer.on("add", (obj: any) => {
    console.log(`Added: ${obj.metadata!.name}`);
  });
  informer.on("update", (obj: any) => {
    console.log(`Updated: ${obj.metadata!.name}`);
  });
  informer.on("delete", (obj: any) => {
    console.log(`Deleted: ${obj.metadata!.name}`);
  });

  await informer.start();
```
Code above works as expected for about the first 10-15 mins, but stops receiving updates for config maps after that.

### PR DESCRIPTION
**Pull request by brendandburns**
May help fix #364 

cc @mag415

### CODE REVIEW

File: src/watch.ts [hunk 1]
```diff
@@ -64,6 +64,12 @@ export class Watch {
                 // ignore parse errors
             }
         });
+        stream.on('error', (err) => {
+            done(err);
+        });
+        stream.on('close', () => {
+            done(null);
```
Comment (drubin) [inline src/watch.ts]: is this related to it closing on the serverside? if so I think we should be passing a custom error object to the `done()` function

Comment (brendandburns) [reply src/watch.ts]: This would be a server-side close, but I think that's an expected case? I don't think we want to return an error there.

**Comment by brendandburns** [comment]
Unit tests added (and they fail w/o the code changes) so this is definitely a bug.

@drubin please take another look. Thanks!

**Comment by drubin** [comment]
/lgtm

Thanks for adding the tests!
